pages/[page].tsx

```diff
@@ -1,33 +1,19 @@
 import { GetStaticProps } from 'next';
 import { useRouter } from 'next/router';
 import React from 'react';
-import Layout from '../components/Layout';
-import styles from '../styles/Page.module.scss';
-import {Link, pageQuery, PageProps, fetchContentful} from '../utils';
+import PageTemplate from '../components/PageTemplate';
+import {pageQuery, PageProps, fetchContentful} from '../utils';
 
 export default function Page(props: PageProps): JSX.Element {
   const {pageName, links, redirect} = props;
   const router = useRouter();
 
-  if (redirect || (typeof window !== 'undefined')) {
+  if (redirect || typeof window !== 'undefined') {
     void router.push('/');
   }
 
   return (
-    <Layout
-      content={(
-        <div id={styles.container}>
-          <h1>{pageName}</h1>
-          <ul>
-            {links && links.map(({displayName, url}: Link, i) => (
-              <li key={i}>
-                <a rel="noreferrer" target='_blank' href={url} key={i}>{displayName}</a>
-              </li>
-            ))}
-          </ul>
-        </div>
-      )}
-    />
+    <PageTemplate pageName = {pageName} links = {links} redirect = {redirect} />
   );
 }
```

Comment:
ideally, all of this should be inside page template!

if your concern is how to handle the redirecting, then I would utilize the pathname as a condition to redirect (i.e. `router.pathname !=== '/' && redirect`